src/main/scala/com/databricks/spark/csv/DefaultSource.scala

```diff
@@ -49,7 +50,7 @@ class DefaultSource
       parameters: Map[String, String],
       schema: StructType) = {
     val path = checkPath(parameters)
-    val delimiter = parameters.getOrElse("delimiter", ",")
+    val delimiter = StringEscapeUtils.unescapeJava(parameters.getOrElse("delimiter", ","))
```

Comment:
I prefer the more general approach. There are some crazy delimiters out there in the wild!